### PR TITLE
set scopes to optional for compute instance

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/8871

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
compute: set optional to `scopes` on `google_compute_instance`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go [hunk 1]
```diff
@@ -800,6 +800,37 @@ func TestAccComputeInstance_serviceAccount(t *testing.T) {
...
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstance_serviceAccount_update1(instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeInstanceExists(
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go:817]: An import check should supersede the exists check.

**Review by rileykarson — APPROVED** [review]

**Review by rileykarson — CHANGES_REQUESTED** [review]
Actually, I think we want to ensure that users are still able to set `scopes = []` explicitly. AFAIK it was a breaking change to add `MinItems`, so we'll want to revert it. Was there a permadiff before or something?

**Comment by edwardmedia** [comment]
@rileykarson the use case in https://github.com/hashicorp/terraform-provider-google/issues/8792#issue-844300180 leads to crash. Set MinItems was intended to address that.

**Comment by rileykarson** [comment]
I'm still not certain this is the right change, I think the actual problem is in Read- it reads like we're setting `service_account` to `nil` when we want to set it to an empty map. This change makes it possible to define the following block. Can you add a test doing so and ensure that that works as expected, provisioning an instance with no service account and no scopes?

```
service_account {}
```

As raised in #8871 I believe that the following definition should work, and removing the ability to specify it was a breaking change that we should revert.

```
  service_account {
    email  = "test-service-account@test-project.iam.gserviceaccount.com" # or some other SA
    scopes = []
  }
```

**Comment by edwardmedia** [comment]
@rileykarson adding below block seems to be fine. Since `email` is optional, should we support the block without the `email`?
```
 service_account {
    email  = "test-service-account@test-project.iam.gserviceaccount.com" # or some other SA
    scopes = []
  }
```

**Pull request closed without merging**